layout-server/src/test/java/org/finos/vuu/layoutserver/controller/LayoutControllerTest.java

```diff
@@ -0,0 +1,165 @@
+    private LayoutRequestDTO layoutRequest;
+    private LayoutResponseDTO expectedLayoutResponse;
+    private List<MetadataResponseDTO> expectedMetadataResponse;
+
+    @BeforeEach
+    public void setup() {
+        validLayoutId = UUID.randomUUID();
+        doesNotExistLayoutId = UUID.randomUUID();
+        UUID metadataId = UUID.randomUUID();
+        String layoutDefinition = "Test Definition";
+
+        baseMetadata = new BaseMetadata();
+        baseMetadata.setName("Test Layout");
+        baseMetadata.setUser("Test User");
+        baseMetadata.setGroup("Test Group");
+        baseMetadata.setScreenshot("Test Screenshot");
+
+        metadata = new Metadata();
+        metadata.setId(metadataId);
+        metadata.setBaseMetadata(baseMetadata);
+
+        layout = new Layout();
+        layout.setId(validLayoutId);
+        layout.setDefinition(layoutDefinition);
+        layout.setMetadata(metadata);
+
+        layoutRequest = new LayoutRequestDTO();
+        MetadataRequestDTO metadataRequestDTO = new MetadataRequestDTO();
+        metadataRequestDTO.setBaseMetadata(baseMetadata);
+        layoutRequest.setDefinition(layout.getDefinition());
+        layoutRequest.setMetadata(metadataRequestDTO);
+
+        expectedLayoutResponse = new LayoutResponseDTO();
+        expectedLayoutResponse.setId(layout.getId());
+        expectedLayoutResponse.setDefinition(layout.getDefinition());
+
+        MetadataResponseDTO metadataResponse = getMetadataResponseDTO();
+        expectedLayoutResponse.setMetadata(metadataResponse);
+
+        expectedMetadataResponse = new ArrayList<>();
+        expectedMetadataResponse.add(metadataResponse);
+    }
+
+
+    @Test
+    void getLayout_layoutExists_returnsLayout() {
+        when(layoutService.getLayout(validLayoutId)).thenReturn(layout);
+        when(modelMapper.map(layout, LayoutResponseDTO.class)).thenReturn(
+            expectedLayoutResponse);
+        assertThat(layoutController.getLayout(validLayoutId)).isEqualTo(expectedLayoutResponse);
+    }
+
+    @Test
+    void getLayout_layoutDoesNotExist_throwsNotFoundAndReturns404() {
+        when(layoutService.getLayout(doesNotExistLayoutId)).thenThrow(NoSuchElementException.class);
+        assertThrows(NoSuchElementException.class,
+            () -> layoutController.getLayout(doesNotExistLayoutId));
+    }
+
+    @Test
+    void getMetadata_metadataExists_returnsMetadata() {
+        when(metadataService.getMetadata()).thenReturn(List.of(metadata));
+        when(modelMapper.map(metadata, MetadataResponseDTO.class)).thenReturn(
+            getMetadataResponseDTO());
+
+        assertThat(layoutController.getMetadata()).isEqualTo(expectedMetadataResponse);
+    }
+
+    @Test
+    void getMetadata_noMetadataExists_returnsEmptyArray() {
+        when(metadataService.getMetadata()).thenReturn(List.of());
+        assertThat(layoutController.getMetadata()).isEmpty();
+    }
+
+    @Test
+    void createLayout_validLayout_returnsCreatedLayout() {
```

Comment:
should we add a test case for createdLayout resulting in 400? i.e. when description is ""

Reply:
That's a test case in `LayoutIntegrationTest`. Do you think it's beneficial to cover this scenario in a unit test as well?

Reply:
makes sense to cover it in integration tests

Reply:
[45ff964](https://github.com/ScottLogic/finos-vuu/pull/68/commits/45ff96496048a1aee5b42cd1547c8e50e1f8a1ac)